File: JS/coffee/terminal.js
import { twoDigit } from "../util/util.js";

const UNIT_PRICE = "unitPrice";

export const payOne = (methodPayment, coffee, callbackStrategy) => {
    callbackStrategy(methodPayment, twoDigit(coffee.price));
}


export const payAll = (methodPayment, coffees = [], callbackStrategy) => {
    let amount = coffees.reduce((a, b) => a + b.price, 0);
    callbackStrategy(methodPayment, twoDigit(amount));
}

File: Java/src/main/java/coffees/Terminal.java
package coffees;

import java.util.List;

public class Terminal {
    public boolean payOne(Coffee coffee, Payment paymentStrategy){
        if(paymentStrategy==null){
            throw new IllegalArgumentException("There is no payment method");
        }
        return paymentStrategy.transfer(coffee.getPrice());
    }

    public boolean payAll(List<Coffee> coffees, Payment paymentStrategy){
        if(paymentStrategy==null){
            throw new IllegalArgumentException("There is no payment method");
        }
        return paymentStrategy.transfer(coffees.stream().mapToDouble(Coffee::getPrice).sum());
    }
}
